cloudant-client/src/test/java/com/cloudant/tests/CloudantClientTests.java

```diff
@@ -464,6 +464,24 @@ public void testBasicAuth() throws IOException {
         client.getAllDbs();
     }
 
+    /**
+     * Test that configuring the Basic Authentication interceptor from credentials and adding to
+     * the CloudantClient works.
+     */
+    @Test
+    @Category(RequiresCloudant.class)
+    public void testBasicAuthFromCredentials() throws IOException {
+        BasicAuthInterceptor interceptor =
+                BasicAuthInterceptor.createFromCredentials(CloudantClientHelper.COUCH_USERNAME,
+                        CloudantClientHelper.COUCH_PASSWORD);
+
+        CloudantClient client = ClientBuilder.account(CloudantClientHelper.COUCH_USERNAME)
+                .interceptors(interceptor).build();
+
+        // Test passes if there are no exceptions
+        client.getAllDbs();
+    }

```

Comment:
Could we make this test run in all envs by using the `MockWebServer` to make sure the creds are added to the request?